File: arrays/check_if_all_1s_are_at_least_length_k_places_away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        count = k  # Initialize to k to handle first 1
        
        for n in nums:
            if n == 1:
                if count < k:
                    return False
                count = 0
            else:
                count += 1
                
        return True

File: arrays/check_if_all_1s_are_at_least_length_k_places_away.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Boolean}
def k_length_apart(nums, k)
    count = k
    
    nums.each do |n|
        if n == 1
            return false if count < k
            count = 0
        else
            count += 1
        end
    end
    
    true
end